### PR TITLE
[INSTALL, Makefile] Use BSD/GNU compatible options

### PR DESCRIPTION
**Pull request by artagnon**
Avoid unnecessary dependency on coreutils and gnu-sed on macOS. The hunk
in src/Makefile.config is non-trivial, as there is no known method to
replace a single line by multiple lines in sed, in a compatible way; but
it was found that this replacement is completely unnecessary, as
evidenced by the built fstar being able to check the entire ulib.

Signed-off-by: Ramkumar Ramachandra <artagnon@gmail.com>

### CODE REVIEW

**Comment by mtzguido** [comment]
This worked well for me in Linux, but we should test in OSX. Thanks!

**Comment by artagnon** [comment]
Works for me on macOS.

File: src/Makefile.config [hunk 1]
```diff
@@ -1,42 +1,37 @@
...
+MSBUILD := $(MSBUILD) /verbosity:minimal /p:Configuration=$(CONFIGURATION)
+DOS2UNIX=$(shell which dos2unix >/dev/null 2>&1 && echo dos2unix || echo true)
+
+# Use options compatible between BSD and GNU versions, on macOS and Linux
+HEAD=head
+SED=sed
+FIND=find
+
+# --------------------------------------------------------------------
+
```
Comment (catalin-hritcu) [inline src/Makefile.config]: Can we get a cleaner diff here? Or is there really a need to change the line endings in this file?

Comment (artagnon) [reply src/Makefile.config]: I thought the official policy was to use LF endings, not CRLF ones; I can revert if necessary.

Comment (catalin-hritcu) [reply src/Makefile.config]: Okay, it seems that it's indeed the policy :)
https://github.com/FStarLang/FStar/blob/master/CONTRIBUTIONS.md#line-endings

For future larger diffs I would still suggest separating line ending changes from the real changes one needs to review.

**Comment by artagnon** [comment]
Replaced by #1738.

**Pull request closed without merging**